src/main/java/com/xceptance/neodymium/NeodymiumRunner.java

```diff
@@ -202,6 +237,16 @@ protected Object createTest() throws Exception
             testMethods.addAll(buildCrossProduct(testAnnotatedMethod.getMethod(), builderList, builderDataList));
         }
 
+        // filter test methods by regex
+        String testExecutionRegex = Neodymium.configuration().getTestNameFilter();
+        if (testExecutionRegex != null)
+        {
+            testMethods = testMethods.stream()
+                                     .filter(testMethod -> (testMethod.getMethod().getDeclaringClass() + "#"
```

Comment:
as discussed in the meeting please use a (GLOB) pattern approach:

```
.filter(testMethod -> {
                                         String functionName = testMethod.getMethod().getDeclaringClass().getName() + "#"
                                                               + testMethod.getName();
                                         return Pattern.compile(testExecutionRegex)
                                                       .matcher(functionName)
                                                       .find();
                                     })
```

Reply:
you can adjust the testNameFilter for the test cases afterwards to:
```
     properties.put("neodymium.testNameFilter",
                       "TestCaseFiltering#(shouldBeExecuted|shouldBeExecutedForDataSetWithExecutableId) :: executable");
```

Reply:
Done, thanks for the advice. I haven't thought about that!